config/routes.rb

```diff
@@ -46,14 +43,22 @@
 
     resources :invitations
     resources :users, only: [ :index, :show, :edit, :update ] do
-      # Trying out the pattern talked about here
-      # http://jeromedalbert.com/how-dhh-organizes-his-rails-controllers/
       resources :password, only: [ :create ], module: "users"
     end
 
     resources :bookmarks
   end
 
+  namespace :api do
+    namespace :v1 do
+      # Disable having to .json the request url but default to JSON. Don't
+      # include the new and edit routes that normal html routes expect
+      scope format: false, except: [ :new, :edit ], defaults: { format: :json } do
```

Comment:
DSL is weird but it seems like a good set of defaults

Reply:
Yeah it got weird real quick, but I've accepted that as a feature of rails routes definitions